### PR TITLE
add steps for Apple Silicon to the web setup

### PR DESCRIPTION
**Pull request by gmanchon**
# context

I just put into shape the steps from @progapandist in this message : https://lewagon-alumni.slack.com/archives/G02NFDT0J/p1610394683060100

these basic steps are the ones that we use for the data setup and that work pretty well

then I tested on a brand new M1... and for the moment everything seems to be working correctly 👌

# options

## cpu architecture detection opt 1

this first solution to test the context (arm vs x86 vs rosetta) is not ideal as it does not detect rosetta, but is enough for the beginning of the setup when the Terminal app runs without Rosetta

``` bash
if [[ `uname -m` == 'arm64' ]]; then
  echo "Your computer uses Apple Silicon 🌟"
else
  echo "Your computer has an Intel processor 🤖"
fi
```

## cpu architecture detection opt 2

this [solution](https://cutecoder.org/software/detecting-apple-silicon-shell-script/) is a little more exhaustive but pretty verbose :

``` bash
arch_name="$(uname -m)"

if [ "${arch_name}" = "x86_64" ]; then
    if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]; then
        echo "Your computer uses Apple Silicon (Rosetta) 🌟"
    else
        echo "Your computer has an Intel processor 🤖"
    fi
elif [ "${arch_name}" = "arm64" ]; then
    echo "Your computer uses Apple Silicon 🌟"
else
    echo "Unknown architecture: ${arch_name}, call a TA 🤔"
fi
```

### CODE REVIEW

**Comment by gmanchon** [comment]
Same content as https://github.com/lewagon/setup/pull/262

**Comment by gmanchon** [comment]
I only created and ran an empty rails project... Do you have other ideas for the validation ?

**Comment by krokrob** [comment]
@gmanchon who should review that PR?

**Comment by barangerbenjamin** [comment]
image

Heya 👋 I get an error but still get an output ☝️ Also I'm running an AMD processor, not an Intel. Don't know if we want to be specific. Maybe it could just be `running M1 or not` if the instructions are the same otherwise.

**Comment by gmanchon** [comment]
@barangerbenjamin good point 👍, I guess this script only makes sense when running on a Mac since all the messages are incorrect in any other context...

I thought Silicon vs Intel would be more difficult for a student to get wrong, as opposed to missing a **not** in the message, wdyt ?

**Comment by gmanchon** [comment]
@barangerbenjamin following a suggestion by @ssaunier I moved the script to a dedicated file in utils/ called through curl.

This will also make it easier to change it in the future if we wish to make something a little fancier that works on all platforms.

**Comment by gmanchon** [comment]
Adresses https://github.com/lewagon/setup/pull/262

File: _partials/osx_oh_my_zsh.md [hunk 1]
```diff
@@ -32,3 +32,27 @@ On Mac, open `Terminal > Preferences` and set the "Pro" theme as default in `Pro
 **Quit** and restart the Terminal. It should now have a nice black background, more easy on the eyes.
 
 :bulb: There are plenty of themes available on the Internet like [MaterialDark](https://github.com/lysyi3m/macos-terminal-themes#materialdark-download) if you fancy trying another one. That's something you can configure later during the day or come back to it if you are done with your setup early. Please carry on with the Github setup!
+
+## Apple Silicon computers
+
+<details>
+  <summary>Forgot if your computer uses Apple Silicon ?</summary>
```
Comment (barangerbenjamin) [inline _partials/osx_oh_my_zsh.md:39]: There is no space between word and punctuation in English 👍 Should be `Silicon?`

**Review by barangerbenjamin — CHANGES_REQUESTED** [review]
Just punctuation related, otherwise LGTM 🙌 🙏

File: _partials/osx_oh_my_zsh.md (hunk 1 above)
Comment (gmanchon) [inline _partials/osx_oh_my_zsh.md:39]:
```suggestion
  <summary>Forgot if your computer uses Apple Silicon?</summary>
```

Comment (gmanchon) [reply _partials/osx_oh_my_zsh.md]: Done 👍 🙏

**Comment by gmanchon** [comment]
Thanks @barangerbenjamin  🔥

**Pull request merged**